task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -425,7 +443,14 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         )
 
         # TODO: Run task failure callbacks here
-    except (AirflowTaskTimeout, AirflowException):
+    except AirflowTaskTimeout:
+        # We should allow retries if the task has defined it
+        msg = TaskState(
+            state=TerminalTIState.FAILED,
+            end_date=datetime.now(tz=timezone.utc),
+        )
+        # TODO: Run task failure callbacks here
```

Comment:
Might conflict with https://github.com/apache/airflow/pull/45308. Will handle it once the first of these 2 is merged